tx_check.go

```diff
@@ -13,7 +13,16 @@ import (
 // because of caching. This overhead can be removed if running on a read-only
 // transaction, however, it is not safe to execute other writer transactions at
 // the same time.
-func (tx *Tx) Check(kvStringer KVStringer) <-chan error {
+func (tx *Tx) Check() <-chan error {
+	return tx.CheckWithStringer(nil)
+}
+
+// CheckWithStringer allows users to provide a customized `KVStringer` implementation,
+// so that bolt can generate human-readable diagnostic messages.
+func (tx *Tx) CheckWithStringer(kvStringer KVStringer) <-chan error {
```

Comment:
Thank you. It's +- what I wanted to do: 

How about more generic: 

```
type CheckOptions struct{
   KeyValueStringer KVStringer
} 

CheckWithOptions(options CheckOptions) {...}
```

This will enable API compatibility going forward without the need to introduce additional methods.

Reply:
I agree it's more flexible. The only minor concern is that it might be a little over engineering for now. 

Updated & thx.

Reply:
I am curious why we don't update the existing method `(*Tx) Check()` ? Why do we need to add a new one CheckWithOptions? 

```
func (tx *Tx) Check(options ...CheckOption) <-chan error
```

Reply:
How about this https://github.com/etcd-io/bbolt/pull/427?